baseapp/abci.go

```diff
@@ -53,6 +53,7 @@ func (app *BaseApp) InitChain(req abci.RequestInitChain) (res abci.ResponseInitC
 	// initialize the deliver state and check state with a correct header
 	app.setDeliverState(initHeader)
 	app.setCheckState(initHeader)
+	app.chainID = req.ChainId
```

Comment:
cc @cmwaters 

> How does the SDK handle the Chain ID part?

Reply:
I meant usually? Does baseapp have access to the chainID in mainline SDK

Reply:
ahh I see, they pass it in a weird way using options and reloading the genesis during initiallization of the server

https://github.com/cosmos/cosmos-sdk/blob/666c345ad23ddda9523cc5cd1b71187d91c26f34/server/util.go#L460-L469

Reply:
figured just setting it during init chain is very similar and simpler

Reply:
Ok